front/src/pages/companies/CompanyShow.tsx

```diff
@@ -31,11 +32,15 @@ export function CompanyShow() {
   const companyId = useParams().companyId ?? '';
   const { insertCompanyFavorite, deleteCompanyFavorite } = useFavorites();
 
+  const navigate = useNavigate();
   const theme = useTheme();
   const { data } = useCompanyQuery(companyId);
   const company = data?.findUniqueCompany;
 
-  if (!company) return <></>;
+  if (!company) {
```

Comment:
The navigate should be put in a useEffect, also check that query is not loading before navigating.

The return <></> can be kept separate from the useEffect

Reply:
Something like this? `  useEffect(() => {
    if (!loading && !person) {
      navigate(AppPath.NotFound);
    }
  }, [loading, person, navigate]);`